The French Dicollecte project website has vanished from the internet. This patch replaces
Dicollecte with a similar project called Grammalecte. And of course, hunspell won't build without
_all_ the languages working. This is great of course, because **I don't even know french**.

Unfortunately, Grammalecte doesn't have archives for version 5.3 which means this patch must also
bump the version to 7.7. This introduces other issues, like the fact that version 7.7 no longer
contains the fr-moderne files.

Why did the French have to do this to me!?

There are other minor issues with other language pack sources, however, those are trivial.

diff --git a/pkgs/by-name/hu/hunspell/dictionaries.nix b/pkgs/by-name/hu/hunspell/dictionaries.nix
index 9ca7e0e7a303..b847086e3f36 100644
--- a/pkgs/by-name/hu/hunspell/dictionaries.nix
+++ b/pkgs/by-name/hu/hunspell/dictionaries.nix
@@ -113,8 +113,8 @@ let
       _name = "ooo_swedish_dict_${_version}";
       readmeFile = "LICENSE_en_US.txt";
       src = fetchurl {
-        url = "https://extensions.libreoffice.org/extensions/swedish-spelling-dictionary-den-stora-svenska-ordlistan/${version}/@@download/file/${_name}.oxt";
-        sha256 = "b982881cc75f5c4af1199535bd4735ee476bdc48edf63e3f05fb4f715654a7bc";
+        url = "https://extensions.libreoffice.org/assets/downloads/z/ooo-swedish-dict-${_version}.oxt";
+        hash = "sha256-uYKIHMdfXErxGZU1vUc17kdr3Ejt9j4/BftPcVZUp7w=";
       };
       meta = {
         longDescription = ''
@@ -146,7 +146,7 @@ let
       '';
     };
 
-  mkDictFromDicollecte =
+  mkDictFromGrammalecte =
     {
       shortName,
       shortDescription,
@@ -156,17 +156,17 @@ let
     }:
     mkDict rec {
       inherit dictFileName;
-      version = "5.3";
-      pname = "hunspell-dict-${shortName}-dicollecte";
+      version = "7.7";
+      pname = "hunspell-dict-${shortName}-grammalecte";
       readmeFile = "README_dict_fr.txt";
       src = fetchurl {
-        url = "http://www.dicollecte.org/download/fr/hunspell-french-dictionaries-v${version}.zip";
-        sha256 = "0ca7084jm7zb1ikwzh1frvpb97jn27i7a5d48288h2qlfp068ik0";
+        url = "https://www.grammalecte.net/dic/hunspell-french-dictionaries-v${version}.zip";
+        sha256 = "sha256-RDFNmS+UtGWMMahu8jUXJKQwZ1MbCvNkP5G/AiDu5hY=";
       };
       meta = {
         inherit longDescription;
-        description = "Hunspell dictionary for ${shortDescription} from Dicollecte";
-        homepage = "https://www.dicollecte.org/home.php?prj=fr";
+        description = "Hunspell dictionary for ${shortDescription} from Grammalecte";
+        homepage = "https://www.grammalecte.net";
         license = lib.licenses.mpl20;
         maintainers = with lib.maintainers; [ renzo ];
         platforms = lib.platforms.all;
@@ -628,7 +628,7 @@ rec {
 
   # FRENCH
 
-  fr-any = mkDictFromDicollecte {
+  fr-any = mkDictFromGrammalecte {
     shortName = "fr-any";
     dictFileName = "fr-toutesvariantes";
     shortDescription = "French (any variant)";
@@ -638,7 +638,7 @@ rec {
     '';
   };
 
-  fr-classique = mkDictFromDicollecte {
+  fr-classique = mkDictFromGrammalecte {
     shortName = "fr-classique";
     dictFileName = "fr-classique";
     shortDescription = "French (classic)";
@@ -649,19 +649,7 @@ rec {
     '';
   };
 
-  fr-moderne = mkDictFromDicollecte {
-    shortName = "fr-moderne";
-    dictFileName = "fr-moderne";
-    shortDescription = "French (modern)";
-    longDescription = ''
-      Ce dictionnaire propose une sélection des graphies classiques et
-      réformées, suivant la lente évolution de l’orthographe actuelle. Ce
-      dictionnaire contient les graphies les moins polémiques de la réforme.
-    '';
-    isDefault = true;
-  };
-
-  fr-reforme1990 = mkDictFromDicollecte {
+  fr-reforme1990 = mkDictFromGrammalecte {
     shortName = "fr-reforme1990";
     dictFileName = "fr-reforme1990";
     shortDescription = "French (1990 reform)";
@@ -708,11 +696,11 @@ rec {
     srcs = [
       (fetchurl {
         url = "http://xuxen.eus/static/hunspell/eu_ES.aff";
-        sha256 = "12w2j6phzas2rdzc7f20jnk93sm59m2zzfdgxv6p8nvcvbrkmc02";
+        sha256 = "1dhiy77nqlnz7q7a628lacp7wwlx07qzcbjwc4a7nclmimknfzrf";
       })
       (fetchurl {
         url = "http://xuxen.eus/static/hunspell/eu_ES.dic";
-        sha256 = "0lw193jr7ldvln5x5z9p21rz1by46h0say9whfcw2kxs9vprd5b3";
+        sha256 = "01mb5qby5fpnh3i7mm0x6i8mzgzqs8368v554vmhrx4kxh11r6cb";
       })
     ];
   };
